clrdefinitions.cmake

```diff
@@ -10,6 +10,12 @@ if (PRERELEASE)
   # add_definitions(-DFEATURE_XXX)
 endif (PRERELEASE)
 
+add_compile_definitions($<$<BOOL:$<TARGET_PROPERTY:DAC_COMPONENT>>:DACCESS_COMPILE>)
```

Comment:
The `CROSSGEN_COMPONENT` and `DAC_COMPONENT` target properties are meant to replace the `crossgen.cmake` and `dac.cmake` files that were littered throughout the repository. The idea is that if a target is one of the "flavors" of a library that gets merged into the DAC or crossgen, we have these flags automatically light up. These flags aren't used for individual targets (for example the `DAC_COMPONENT` one is used by 13 targets). Since these are all across the repository (utilcode, each metadata component, `cee_*` targets, cordbee, strongname_api, unwinder), I felt it best to centralize them instead of copying them all over the repository.

The problem with `remove_definitions` is that it doesn't exist in a target-specific form.

The reasoning behind switching these to be target-focused instead of directory-focused was to enable removing a lot of the `subfolder/CMakeLists.txt` files where the CMakeLists.txt was the only file in the folder. For example, this system enables us to define all of the utilcode targets in a single CMakeLists.txt file instead of having four subfolders only containing CMakeLists.txt files.

Reply:
I actually like removing of the leaf folders generating variants of the same stuff with slightly different options. I have to admit I was a bit skeptical when we've discussed it in the past, but the result looks great.
I have not realized that we actually have multiple targets all being part the DAC_COMPONENT, so my idea would result in something ugly like the dac.cmake or crossgen.cmake even if it worked.
So let's keep it the way you have it.